### ISSUE TITLE
operator spread flattens Channel

### ISSUE THREAD

**Issue by MaxUlysse** [nextflow-io/nextflow#241]
```
Channel
	.from( ['a','z', file('a')], ['a','z', file('a')], ['b', 'q', file('b')], ['b', 'q', file('b')])
	.groupTuple(by:[0,1])
	.println()
```
Returns:
```
[a, z, [/path/a, /path/a]]
[b, q, [/path/b, /path/b]]
```
Now I'm using spread to add another value to the Channel
```
A = Channel
	.from( ['a','z', file('a')], ['a','z', file('a')], ['b', 'q', file('b')], ['b', 'q', file('b')])
	.groupTuple(by:[0,1])

Channel
	.from('X')
	.spread(A)
	.println()

```
Returns:
```
[X, a, z, /path/a, /path/a]
[X, b, q, /path/b, /path/b]
```
But the expected output should be:
```
[X, a, z, [/path/a, /path/a]]
[X, b, q, [/path/b, /path/b]]
```

**Comment by mes5k** [comment]
Even if this is as designed, it would be great if there was an option to achieve the don't-completely-flatten behavior described above.

**Comment by pditommaso** [comment]
It makes sense. The problem that changing the semantic would break the existing code. I can add an option `flat`, default `true`, that when turned off implements the suggested behaviour.

**Comment by pditommaso** [comment]
@MaxUlysse I've implemented this, though now your snippet returns: 

```
[X, [a, z, [/path/a, /path/a]]]
[X, [b, q, [/path/b, /path/b]]]
```

That in my opinion is coherent. Thoughts ?

**Comment by pditommaso** [comment]
cc @emi80

**Comment by pditommaso** [comment]
I've you want to give a try to it you can use `NXF_VER=0.23.4-SNAPSHOT`

**Comment by MaxUlysse** [comment]
It is indeed better as it's not flattening the channel anymore.
Thanks a lot.

**Comment by pditommaso** [comment]
Resolved by #298.
